query-connector/src/app/queryBuilding/components/ConditionSelection.tsx

```diff
@@ -0,0 +1,201 @@
+
+type ConditionSelectionProps = {
+  fetchedConditions: CategoryNameToConditionOptionMap;
+  selectedConditions: CategoryNameToConditionOptionMap;
+  setBuildStep: (buildStep: BuildStep) => void;
+
+  setFetchedConditions: Dispatch<
+    SetStateAction<CategoryNameToConditionOptionMap | undefined>
+  >;
+  setSelectedConditions: Dispatch<
+    SetStateAction<CategoryNameToConditionOptionMap | undefined>
+  >;
+  queryName: string;
+  validateForm: () => void;
+  setFormError: Dispatch<SetStateAction<FormError>>;
+  formError: FormError;
+  setLoading: Dispatch<SetStateAction<boolean>>;
+  loading: boolean;
+  setConditionValueSets: Dispatch<
+    SetStateAction<ConditionIdToValueSetArray | undefined>
+  >;
+};
+
+/**
+ * Display component for a condition on the query building page
+ * @param root0 - params
+ * @param root0.fetchedConditions - ID of the condition to reference
+ * @param root0.selectedConditions - name of condition to display
+ * @param root0.setBuildStep - Redirect function to handle view routing
+ * @param root0.setFetchedConditions - listener function for checkbox
+ * selection
+ * @param root0.setSelectedConditions - current checkbox selection status
+ * @param root0.queryName - current checkbox selection status
+ * @param root0.validateForm - function that checks for a valid query name
+ * and at least one selected condition
+ * @param root0.formError - indicates missing or incorrect form data
+ * @param root0.setFormError - state function that updates the status of the
+ * condition selection form input data
+ * @param root0.loading -  Boolean to track loading state
+ * @param root0.setLoading - The function to set the loading state
+ * @param root0.setConditionValueSets - state function that updates the status of the
+ * condition selection form input data
+ * @returns A component for display to redner on the query building page
+ */
+export const ConditionSelection: React.FC<ConditionSelectionProps> = ({
+  fetchedConditions,
+  selectedConditions,
+  setBuildStep,
+  setFetchedConditions,
+  setSelectedConditions,
+  queryName,
+  validateForm,
+  formError,
+  setFormError,
+  loading,
+  setConditionValueSets,
+  setLoading,
+}) => {
+  const focusRef = useRef<HTMLInputElement | null>(null);
+
+  const [searchFilter, setSearchFilter] = useState<string>();
+
+  useEffect(() => {
+    let isSubscribed = true;
+
+    if (queryName == "" || queryName == undefined) {
+      focusRef?.current?.focus();
+    }
+
+    async function fetchConditionsAndUpdateState() {
+      const { categoryToConditionArrayMap } = await getConditionsData();
+
+      if (isSubscribed) {
+        setFetchedConditions(
+          mapFetchedDataToFrontendStructure(categoryToConditionArrayMap),
+        );
+      }
+    }
+
+    fetchConditionsAndUpdateState().catch(console.error);
+    return () => {
+      isSubscribed = false;
+    };
+  }, []);
+
+  async function getValueSetsForSelectedConditions() {
```

Comment:
@fzhao99 @robertandremitchell would love any suggestions you have for how to organize the valuesets and concepts per condition here. I tried to use prior art where possible/available, but it's definitely proven to be finicky to access some of the deeply-nested properties (see: batch selecting/unselecting all value sets and concepts for a given Group)

Reply:
eek think I'm beginning to understand the thorniness of the issues here.

The way I'm digesting this: the category --> condition --> valueset --> concept hierarchy from the data needs to be mixed with the labs / conditions / medications groupings at the valueset level. Using what we have already, a sorted labs/conditions/medications grouping of all selected valuesets for all selected condition needs to be stored in memory (ie to display for the accordion setup on the concept selection page). This setup however would make batch selecting all the valuesets / concepts for a condition complicated because we're storing the inclusion status at the concept / valueset level. 

Is that the crux of the organizational issue that you're describing? Anything I'm missing about how I'm framing the problem above?

Reply:
that's the gist of it, yeah! I did just push some changes that get the individual toggles and batch toggles working though 🤞🏻